src/sas/qtgui/Plotting/SlicerParameters.py

```diff
@@ -153,47 +201,122 @@ def onGeneratePlots(self, isChecked):
-            plotter.slicer.setParamsFromModel()
-        pass
+            # Apply plotter to a plot
+            self.applyPlotter(plot)
+            # Save 1D plots if required
+            plots.append(plot)
+        if self.isSave:
+            self.save1DPlotsForPlot(plots)
+        pass  # debug anchor
+
+    def applyPlotter(self, plot):
+        """
+        Apply the current slicer to a plot
+        """
+        # don't assign to itself
+        if plot == self.parent.data[0].name:
+            return
+        # a plot might have been deleted
+        if plot not in self.active_plots:
+            return
+        # get the plotter2D instance
+        plotter = self.active_plots[plot]
+        # Assign model to slicer
+        index = self.cbSlicer.currentIndex()
+
+        slicer = self.callbacks[index]
+        if slicer is None:
+            return
+        plotter.setSlicer(slicer=slicer, reset=False)
+        # override slicer model
+        plotter.slicer._model = self.model
+        # force conversion model->parameters in slicer
+        plotter.slicer.setParamsFromModel()
+
+    def save1DPlotsForPlot(self, plots):
+        """
+        Save currently shown 1D sliced data plots for a given 2D plot
+        """
+        items_for_fit = []
+        for plot in plots:
+            for item in self.active_plots.keys():
+                data = self.active_plots[item].data[0]
+                if not isinstance(data, Data1D):
+                    continue
+                if plot not in data.name:
+                    continue
+                filename = data.name if self.txtExtension.text() == ""\
+                    else data.name + "_" + str(self.txtExtension.text())
+                # Saved as "TXT" file, as in original impl
+                filename_ext = filename + ".txt"
+                GuiUtils.onTXTSave(data, filename_ext)
+                new_item = GuiUtils.createModelItemWithPlot(data, name=filename)
+                self.parent.manager.updateModelFromPerspective(new_item)
+
+                items_for_fit.append(new_item)
+        # Send to fitting, if needed
+        # We can get away with directly querying the UI, since this is the only
+        # place we need that state.
+        fitting_requested = self.cbFitOptions.currentIndex()
+        self.sendToFit(items_for_fit, fitting_requested)
 
     def setModel(self, model):
         """ Model setter """
         self.model = model
         self.proxy.setSourceModel(self.model)
+        self.model.itemChanged.connect(self.onApply)
+
+    def sendToFit(self, items_for_fit, fitting_requested):
+        """
+        Send `items_for_fit` to the Fit perspective, in either single fit or batch mode
+        """
+        if 3 < fitting_requested < 1:
+            return
+        isBatch = fitting_requested == 2
+        # icky way to go up the tree
+        self.parent.manager._perspective().setData(data_item=items_for_fit, is_batch=isBatch)
```

Comment:
This sends data to whichever perspective is currently active.

Reply:
Fitting perspective - best perspective.
Also, well spotted.